db/compaction/compaction_picker_fifo.cc

```diff
@@ -121,20 +121,42 @@ Compaction* FIFOCompactionPicker::PickTTLCompaction(
   return c;
 }
 
+// The size-based compaction picker for FIFO.
+//
+// When the entire column family size exceeds max_table_files_size, FIFO will
+// try to delete the oldest sst file(s) until the resulting column family size
+// is smaller than max_table_files_size.
+//
+// Note that this function also takes care the case where a DB is migrating
+// from level/universal compaction to FIFO compaction.  During the migration,
+// the column family will also have non-L0 files while FIFO can only create L0
+// files. In this case, this function will first purge the sst files in the
+// bottom-most non-empty level first, and the DB will eventually converge to the
+// regular FIFO case where there're only L0 files.
```

Comment:
Maybe worth adding a note that during migration the existing data may not be purged based on it's time. Even the lower level data maybe newer than the high level data.